aws-core/src/test/java/software/amazon/awssdk/awscore/client/builder/DefaultAwsClientBuilderTest.java

```diff
@@ -140,7 +140,7 @@ public void clientFactoryProvided_ClientIsManagedBySdk() {
     public void asyncHttpClientFactoryProvided_ClientIsManagedBySdk() {
         TestAsyncClient client = testAsyncClientBuilder()
             .region(Region.US_WEST_2)
-            .asyncHttpClientBuilder(serviceDefaults -> {
+            .httpClientBuilder((SdkAsyncHttpClient.Builder) serviceDefaults -> {
```

Comment:
we have to cast it here since `SdkDefaultClientBuilder` implements both `SdkAsyncClientBuilder` and `SdkSyncClientBuilder` and has overloads of `httpClientBuilder` for async and sync. I think this is fine as `SdkDefaultClientBuilder` is protected api